q2_fondue/metadata.py

```diff
@@ -35,71 +34,65 @@ def _efetcher_inquire(
     Args:
         efetcher (ef.Efetcher): A valid instance of an Entrezpy Efetcher.
         run_ids (List[str]): List of all the run IDs to be fetched.
+        log_level (str): Logging level.
 
     Returns:
         pd.DataFrame: DataFrame with metadata obtained for the provided IDs.
-        list: List of all the run IDs that were not found.
+        dict: Dictionary of the run IDs that were not found with
+            respective error messages.
     """
     metadata_response = efetcher.inquire(
         {
             'db': 'sra',
             'id': run_ids,
             'rettype': 'xml',
             'retmode': 'xml',
-            'retmax': BATCH_SIZE
+            'retmax': len(run_ids),
+            'reqsize': BATCH_SIZE
```

Comment:
This is how we can actually make the `efetcher` batch the requests. In this way we can (at least theoretically) make use of multi-threading built into `entrezpy`.

Reply:
🤩